src/components/Trade/Trade.tsx

```diff
@@ -10,25 +13,43 @@ export type TradeProps = {
 }
 
 export const Trade = ({ defaultBuyAssetId }: TradeProps) => {
+  const { getDefaultAssets } = useDefaultAssets(defaultBuyAssetId)
+  const [sellAsset, setSellAsset] = useState<Asset | undefined>()
+  const [buyAsset, setBuyAsset] = useState<Asset | undefined>()
+
   const methods = useForm<TS>({
     mode: 'onChange',
     defaultValues: {
       fiatSellAmount: '0',
       fiatBuyAmount: '0',
       amount: '0',
-      sellTradeAsset: { amount: '0' },
-      buyTradeAsset: { amount: '0' },
+      sellTradeAsset: { amount: '0', asset: sellAsset },
+      buyTradeAsset: { amount: '0', asset: buyAsset },
       isExactAllowance: false,
       slippage: 0.002,
     },
   })
 
+  useEffect(() => {
```

Comment:
moving this logic up here allows us to

a) set the default assets on page load
b) set the default assets on routes correctly